src/fidesops/schemas/saas/shared_schemas.py

```diff
@@ -1,5 +1,29 @@
-from typing import Any, Dict, Literal, Optional, Tuple
+from enum import Enum
+from typing import Any, Dict, Optional
+from pydantic import BaseModel
 
 
-SaaSRequestParams = Tuple[Literal["GET", "PUT"], str, Dict[str, Any], Optional[str]]
-"""Custom type to represent a tuple of HTTP method, path, params, and body values for a SaaS request"""
+class HTTPMethod(Enum):
+    """Enum to represent HTTP Methods"""
+
+    GET = "GET"
+    POST = "POST"
+    PATCH = "PATCH"
+    PUT = "PUT"
+    DELETE = "DELETE"
+    OPTIONS = "OPTIONS"
+    HEAD = "HEAD"
+
+
+class SaaSRequestParams(BaseModel):
+    """Custom type to represent a SaaS request param"""
+
+    method: HTTPMethod
+    path: str
+    param: Dict[str, Any]
+    body_values: Optional[str]
```

Comment:
Also nit: should this be `body_value` or just `body` since it's a JSON string that is one value in this context.